lib/banks/savings.rb

```diff
@@ -0,0 +1,39 @@
+require 'pry'
+module BankAccount
+  class SavingsAccount < Account
+    MINIMUM_BALANCE = 10
+    TRANSACTION_FEE = 2
+
+    def initialize(id, initial_balance)
+      super # inherits @id and @current_balance(= initial_balance) from Account class
+      raise ArgumentError.new("Savings accounts\
+       require a minimum balance of $10.") if initial_balance < 10
+    end
+
+    def withdraw(amount)
+      # This method inherits from the Account class #withdraw(amount) method
+      # amount will be checked against current_balance to make sure you have
+      # enough money in your account to make this withdrawl (leaving at least
+      # $10 left in the account)
+      amount = amount + MINIMUM_BALANCE + TRANSACTION_FEE
+      # need to add MINIMUM_BALANCE back on the current_balance
+      super + MINIMUM_BALANCE
+    end
+
+    def rate_as_a_percent(rate)
```

Comment:
zomg this method makes me so happy. Really well done with this. You've identified an edge case; isolated its handling in a reusable method, and provided comments to clarify your intent. :100: A+++++, would review again.